src/plugins/intel_npu/src/plugin/npuw/embedding_infer_request.cpp

```diff
@@ -0,0 +1,265 @@
+#include "openvino/runtime/iasync_infer_request.hpp"
+#include "util.hpp"
+
+ov::SoPtr<ov::ITensor> ov::npuw::EmbeddingInferRequest::create_prefill_output_tensor() {
+    const auto& out_port = m_prefill_request->get_outputs()[0];
+    auto out_shape = out_port.get_shape();
+    auto& kvcache_desc = m_npuw_llm_compiled_model->m_kvcache_desc;
+
+    auto prefill_shape = ov::Shape(out_shape);
+    prefill_shape[layer_ids::INPUT_IDS_SEQ_LEN_DIM] = kvcache_desc.max_prompt_size;
+    return ov::get_tensor_impl(ov::Tensor(out_port.get_element_type(), prefill_shape));
+}
+
+ov::npuw::EmbeddingInferRequest::EmbeddingInferRequest(const std::shared_ptr<LLMCompiledModel>& compiled_model)
+    : ov::npuw::LLMInferBaseRequest(compiled_model) {
+    init_ports();
+
+    m_prefill_request = m_npuw_llm_compiled_model->m_prefill_compiled->create_infer_request();
+
+    for (const auto& input_port : m_prefill_request->get_compiled_model()->inputs()) {
+        m_prefill_in_ports.emplace(input_port.get_any_name(), input_port);
+        // Cache past_key_values ports for efficient clearing
+        if (input_port.get_any_name().find(layer_names::past_key_values) != std::string::npos) {
+            m_prefill_past_kv_ports.push_back(input_port);
+        }
+    }
+    for (const auto& output_port : m_prefill_request->get_compiled_model()->outputs()) {
+        m_prefill_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+
+    m_prefill_output = create_prefill_output_tensor();
+}
+
+void ov::npuw::EmbeddingInferRequest::prepare_for_new_conversation() {
+    namespace uu = ov::npuw::util;
+
+    m_attn_mask_in_tensor = m_prefill_request->get_tensor(m_prefill_in_ports.at(layer_names::attention_mask));
+    m_input_ids_in_tensor = m_prefill_request->get_tensor(m_prefill_in_ports.at(layer_names::input_ids));
+
+    if (auto pos_ids_port = m_prefill_in_ports.find(layer_names::position_ids);
+        pos_ids_port != m_prefill_in_ports.end()) {
+        m_pos_ids_in_tensor = m_prefill_request->get_tensor(m_prefill_in_ports.at(layer_names::position_ids));
+        uu::fill_tensor<int64_t>(m_pos_ids_in_tensor, 0);
+    }
+
+    if (auto type_ids_port = m_prefill_in_ports.find(layer_names::token_type_ids);
+        type_ids_port != m_prefill_in_ports.end()) {
+        m_type_ids_in_tensor = m_prefill_request->get_tensor(m_prefill_in_ports.at(layer_names::token_type_ids));
+        uu::fill_tensor_bytes(m_type_ids_in_tensor, 0);
+    }
+
+    uu::fill_tensor_bytes(m_input_ids_in_tensor, 0u);
+    uu::fill_tensor<int64_t>(m_attn_mask_in_tensor, 0);
+    uu::fill_tensor_bytes(m_prefill_output, 0u);
+
+    // Clear all past_key_values tensors
+    for (const auto& port : m_prefill_past_kv_ports) {
+        uu::fill_tensor_bytes(m_prefill_request->get_tensor(port), 0u);
+    }
+
+    m_npuw_llm_compiled_model->m_kvcache_desc.num_stored_tokens = 0u;
+}
+
+void ov::npuw::EmbeddingInferRequest::infer_chunked_prefill(ov::SoPtr<ov::ITensor> input_ids,
```

Comment:
Is `infer_chunked_prefill` and `infer_whole_prefill` so different for `embedding_infer_request` and `llm_infer_request`?
Can't we have common implementation?
Or at least some common parts?